tests/conftest.py

```diff
@@ -83,6 +83,24 @@ def fixture(request):
     generate_fixture_group(name, variants)
 del generate_fixture_group
 
+# ------------------------------------------------------------------------------
+#                Adding a customizable output dir for plots
+# ------------------------------------------------------------------------------
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--systemtest_outdir", action="store", default="./test_report/generated/plots"
+    )
+
+
+def pytest_generate_tests(metafunc):
+    # This is called for every test. Only get/set command line arguments
+    # if the argument is specified in the list of test "fixturenames".
+    option_value = metafunc.config.option.systemtest_outdir
+    if "systemtest_outdir" in metafunc.fixturenames and option_value is not None:
+        metafunc.parametrize("systemtest_outdir", [option_value])
```

Comment:
Maybe pre-processing this path and ensuring that the directory exists (or creating it if it doesn't) would be a good idea?